### PR TITLE
fix(infra) add support for internal ACPs

### PR DESCRIPTION
**Pull request by dduportal**
Related to https://github.com/jenkins-infra/helpdesk/issues/3954#issuecomment-2114950444

This PR is a medium-term fix to ensure the new "internal" (without public ingress) ACP (in the ci.jenkins.io-agents-1 AKS cluster) can be used without failing the healthcheck.

Notes:
- The configuration (provider names and URL) is already spread between the code here and the puppet templates. This is not the scope of this PR (hence the "middle term") patch.
- The condition is ready to support internal ACPs for future EKS or DOKS using the pattern `(.*)ks-internal` for the name

### CODE REVIEW

**Comment by dduportal** [comment]
Currently testing in https://github.com/jenkinsci/jenkins-infra-test-plugin/pull/126

**Comment by dduportal** [comment]
External test in <> worked as expected:

```
9:24:03  + curl --fail --silent --show-error --location http://artifact-caching-proxy.artifact-caching-proxy:8080/health
19:24:03  OK
19:24:04  INFO: using artifact caching proxy from 'azure-aks-internal' provider.
```

and tested with an existing one (to ensure no regression):

```
19:31:41  + curl --fail --silent --show-error --location https://repo.aws.jenkins.io/health
19:31:41  OK
19:31:41  INFO: using artifact caching proxy from 'aws' provider.
```

File: vars/infra.groovy [hunk 1]
```diff
@@ -194,7 +194,15 @@ Object withArtifactCachingProxy(boolean useArtifactCachingProxy = true, Closure
   // Check if the artifact caching proxy provider is unreachable
   if (useArtifactCachingProxy) {
     boolean healthCheckFailed = false
-    withEnv(["HEALTHCHECK=https://repo.${requestedProxyProvider}.jenkins.io/health"]) {
+    String artifactCachingProxyUrl = "https://repo.${requestedProxyProvider}.jenkins.io/health"
+    // The (.*)Kubernetes-hosted ACP marked as internal uses the Kubernetes internal SVC domain name and custom port as there are no ingress
+    // TODO: retrieve the value from the settings.xml to avoid spreading the URLs here and in puppet:
+    // https://github.com/jenkins-infra/jenkins-infra/blob/15967f693d1845843204d981168e9a847a090ecb/hieradata/common.yaml#L249
+    if(requestedProxyProvider.contains('ks-internal')) {
```
[the remark this answers is not in the document]
Comment (dduportal) [reply vars/infra.groovy:201]: No I meant `ks-internal` willingly as explained in the PR body and the comment

[the remark this answers is not in the document]
Comment (dduportal) [reply vars/infra.groovy:201]: The idea is that the URL is the same for any kubernetes-hosted ACP when using the internal SVC URL, wether it is aks, eks, doks. If we keep using the « ks-internal » convention, eg « Kubernetes Service Internal »

[the remark this answers is not in the document]
Comment (lemeurherve) [reply vars/infra.groovy:201]: > as explained in the PR body

Sorry I missed that!

> the « ks-internal » convention, eg « Kubernetes Service Internal »

Thanks for this explanation! I didn't understand at first why the `ks-` prefix, I though it was only related to the literal end of "aks", "eks", "doks" and initially found it unnecessary.

**Pull request merged**